src/controllers/bulk/bulksupported.h

```diff
@@ -1,18 +1,20 @@
-// A list of supported USB bulk devices
-
 #pragma once
 
-typedef struct bulk_supported {
-    unsigned short vendor_id;
-    unsigned short product_id;
-    unsigned char in_epaddr;
-    unsigned char out_epaddr;
-    unsigned int interface_number;
-} bulk_supported_t;
+#include <cstdint>
+
+// A list of supported USB bulk devices
+
+struct bulk_supported_t {
+    std::uint16_t vendor_id;
```

Comment:
> This "std::" looks untypical because libusb is a c library,

yes, but this on the other hand is C++ code and in C++ code, the fixed width integer types here are in fact in the std namespace.

> We can be sure that the types are matching but they are not the "same"

The have the same bit width and signedness. I would consider them the same.

Reply:
It is OK to keep it here. 
In general It is a good idea to not change types used in a library. This allows us just to include libusb.h without any additional headers and all types are available without implicit casting. In this case this is only, a marginal difference.

Reply:
I agree. In the end these are just type aliases anyway and if we're worried about implicit conversions we should look into `-Wnarrowing`.